nucliadb/tests/ndbfixtures/writer.py

```diff
@@ -0,0 +1,197 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+from typing import AsyncIterator
+from unittest.mock import patch
+
+import pytest
+from fastapi import FastAPI
+from httpx import AsyncClient
+from pytest_lazy_fixtures import lazy_fixture
+from redis import asyncio as aioredis
+
+from nucliadb.writer import tus
+from nucliadb.writer.api.v1.router import KB_PREFIX, KBS_PREFIX
+from nucliadb.writer.app import create_application
+from nucliadb.writer.settings import settings
+from nucliadb_models.resource import NucliaDBRoles
+from nucliadb_utils.settings import (
+    FileBackendConfig,
+    nucliadb_settings,
+    storage_settings,
+)
+from nucliadb_utils.tests.fixtures import get_testing_storage_backend
+from nucliadb_utils.utilities import Utility, clean_utility, set_utility
+from tests.ingest.fixtures import IngestFixture
+
+from .utils import create_api_client_factory
+
+# TODO: replace knowledgebox_ingest for knowledgebox
+
+# Main fixtures
+
+
+@pytest.fixture(scope="function")
+async def component_nucliadb_writer(
+    writer_api_server: FastAPI,
+) -> AsyncIterator[AsyncClient]:
+    client_factory = create_api_client_factory(writer_api_server)
+    async with client_factory(roles=[NucliaDBRoles.WRITER]) as client:
+        yield client
+
+
+@pytest.fixture(scope="function")
+async def nucliadb_writer_manager(
+    nucliadb_writer: AsyncClient,
+) -> AsyncIterator[AsyncClient]:
+    roles = [NucliaDBRoles.MANAGER, NucliaDBRoles.WRITER]
+    nucliadb_writer.headers["X-NUCLIADB-ROLES"] = ";".join([role.value for role in roles])
```

Comment:
Could we use the `client_factory` for this fixture as well? It feels a bit cleaner than manipulating headers directly.

Reply:
I'd prefer not to. Maybe it's not explicit enough but this is using a magic fixture `nucliadb_writer`, that is an AsyncClient to the component, standalone or whatever. It doesn't need to know